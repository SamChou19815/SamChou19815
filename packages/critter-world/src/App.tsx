import React, { ReactElement } from 'react';

import AppBody from './components/App';
import LoginBarrier from './components/LoginBarrier';
import { initializeWindowSizeHooksListeners } from './utils/window-size-hook';

import './index.css';
import './app.css';

if (!__SERVER__) initializeWindowSizeHooksListeners();

const App = (): ReactElement => {
  return (
    <div>
      <nav className="navbar">
        <div className="navbar__inner">
          <div className="navbar__items">
            <a className="navbar__brand" href="/">
              <strong className="navbar__title">Critter World Web UI</strong>
            </a>
          </div>
          <div className="navbar__items navbar__items--right">
            <a className="navbar__item navbar__link" href="https://developersam.com">
              Developer Sam
            </a>
          </div>
        </div>
      </nav>
      <LoginBarrier>
        <AppBody />
      </LoginBarrier>
    </div>
  );
};

export default App;
