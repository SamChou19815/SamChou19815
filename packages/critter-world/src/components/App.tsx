/* eslint-disable no-await-in-loop */
import React, { ReactElement, useState, useEffect } from 'react';

import { sendGetWorldRequest } from '../utils/http';
import DataLoadedApp from './DataLoadedApp';

import LoadingOverlay from 'lib-react/LoadingOverlay';

const wait = (time: number): Promise<void> =>
  new Promise((resolve) => setTimeout(() => resolve(), time));

const recurringRefetchWorldState = (setWorldState: (s: WorldState) => void): (() => void) => {
  let continueToRefresh = true;

  (async () => {
    while (continueToRefresh) {
      try {
        const newWorldState: WorldState = await sendGetWorldRequest();
        setWorldState(newWorldState);
        await wait(30);
      } catch {
        await wait(1000);
      }
    }
  })();

  return (): void => {
    continueToRefresh = false;
  };
};

const App = (): ReactElement => {
  const [worldState, setWorldState] = useState<WorldState | null>(null);
  useEffect(() => recurringRefetchWorldState(setWorldState), []);

  if (worldState == null) return <LoadingOverlay />;
  return <DataLoadedApp worldState={worldState} />;
};

export default App;
