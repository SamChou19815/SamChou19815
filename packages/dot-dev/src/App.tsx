import React, { ReactElement } from 'react';

import FanArtWorkCard from './FanArtWorkCard';
import {
  FAN_ART_ITERATION_0,
  FAN_ART_ITERATION_1,
  FAN_ART_ITERATION_2,
  FAN_ART_ITERATION_3,
  FAN_ART_BIRTHDAY_EDITION,
} from './data';

import initializeThemeSwitching from 'lib-react/theme-switcher-initializer';

import 'infima/dist/css/default/default.min.css';
import './index.css';

if (!__SERVER__) {
  initializeThemeSwitching();
}

const ArtsPage = (): ReactElement => (
  <div>
    <nav className="navbar">
      <div className="navbar__inner">
        <div className="navbar__items">
          <a className="navbar__brand" href="/">
            Fan Arts | Random@dev-sam
          </a>
        </div>
        <div className="navbar__items navbar__items--right">
          <a className="navbar__item navbar__link" href="https://developersam.com">
            Home
          </a>
        </div>
      </div>
    </nav>
    <FanArtWorkCard {...FAN_ART_BIRTHDAY_EDITION} />
    <FanArtWorkCard {...FAN_ART_ITERATION_3} />
    <FanArtWorkCard {...FAN_ART_ITERATION_2} />
    <FanArtWorkCard {...FAN_ART_ITERATION_1} />
    <FanArtWorkCard {...FAN_ART_ITERATION_0} />
  </div>
);

export default ArtsPage;
