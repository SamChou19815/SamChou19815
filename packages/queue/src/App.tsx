import React, { ReactElement } from 'react';

import 'lib-firebase';

import { Provider as ReactReduxProvider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import ConfiguredMainAppBarrier from './components/ConfiguredMainAppBarrier';
import MainApp from './components/MainApp';
import MaterialThemedApp from './components/MaterialThemedApp';
import { store } from './models/redux-store';

const Wrapped = (): ReactElement => (
  <MaterialThemedApp title="Queue">
    <MainApp />
  </MaterialThemedApp>
);

export default (): ReactElement => (
  <BrowserRouter>
    <ReactReduxProvider store={store}>
      <ConfiguredMainAppBarrier appComponent={Wrapped} />
    </ReactReduxProvider>
  </BrowserRouter>
);
