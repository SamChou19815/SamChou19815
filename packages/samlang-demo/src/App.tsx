import React, { ReactElement } from 'react';

import MaterialButtonLink from 'lib-react/MaterialButtonLink';
import MaterialThemedApp from 'lib-react/MaterialThemedApp';

import styles from './App.module.css';
import LanguageDemo from './LanguageDemo';

const buttons: ReactElement = (
  <>
    <MaterialButtonLink href="https://samlang.developersam.com" color="inherit">
      Docs
    </MaterialButtonLink>
    <MaterialButtonLink href="https://developersam.com" color="inherit">
      Home
    </MaterialButtonLink>
  </>
);

const appStyles = {
  app: styles.App,
  title: styles.Title
};

export default (): ReactElement => (
  <MaterialThemedApp styles={appStyles} title="SAMLANG Demo" buttons={buttons}>
    <LanguageDemo />
  </MaterialThemedApp>
);
