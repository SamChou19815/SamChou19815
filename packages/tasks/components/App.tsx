import React, { ReactElement } from 'react';

import Head from 'next/head';
import { Provider as ReactReduxProvider } from 'react-redux';

import { store } from '../models/redux-store';
import HomePage from './pages/HomePage';

export default (): ReactElement => (
  <ReactReduxProvider store={store}>
    <Head>
      <meta charSet="utf-8" />
      <link rel="shortcut icon" href="/favicon.ico" />
      <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
      <meta name="theme-color" content="#F7F7F7" />
      <link rel="manifest" href="/manifest.json" />
      <title>Tasks - Developer Sam Apps</title>
    </Head>
    <HomePage />
  </ReactReduxProvider>
);
