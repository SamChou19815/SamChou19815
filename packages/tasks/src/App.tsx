import React, { ReactElement } from 'react';
import { Provider as ReactReduxProvider } from 'react-redux';
import MaterialThemedApp from 'lib-react/MaterialThemedApp';
import MaterialButtonLink from 'lib-react/MaterialButtonLink';
import LandingPage from './components/pages/LandingPage';
import LoadingPage from './components/pages/LoadingPage';
import MainAppBarrier from './components/util/MainAppBarrier';
import { APP_NAME } from './util/constants';
import styles from './App.module.css';
import { getProjectsObservable, getTasksObservable } from './util/firestore';
import { getPatchProjectsAction, getPatchTasksAction, store } from './models/redux-store';
import RootRouter from './components/pages/RootRouter';

const buttons: ReactElement = (
  <MaterialButtonLink href="https://developersam.com" color="inherit">
    Home
  </MaterialButtonLink>
);

const dataLoader = (): Promise<void> => {
  const projectsObservable = getProjectsObservable();
  const tasksObservable = getTasksObservable();
  let resolvedProjects = false;
  let resolvedTasks = false;
  return new Promise(resolve => {
    projectsObservable.subscribe(({ createdAndEdited, deleted }) => {
      store.dispatch(getPatchProjectsAction(createdAndEdited, deleted));
      if (!resolvedProjects) {
        resolvedProjects = true;
        if (resolvedTasks) {
          resolve();
        }
      }
    });
    tasksObservable.subscribe(({ createdAndEdited, deleted }) => {
      store.dispatch(getPatchTasksAction(createdAndEdited, deleted));
      if (!resolvedTasks) {
        resolvedTasks = true;
        if (resolvedProjects) {
          resolve();
        }
      }
    });
  });
};

const Main = (): ReactElement => (
  <ReactReduxProvider store={store}>
    <RootRouter />
  </ReactReduxProvider>
);

export default (): ReactElement => (
  <MaterialThemedApp
    title={APP_NAME}
    buttons={buttons}
    styles={{ app: styles.App, title: styles.Title }}
  >
    <MainAppBarrier
      dataLoader={dataLoader}
      landingPageComponent={LandingPage}
      loadingPageComponent={LoadingPage}
      appComponent={Main}
    />
  </MaterialThemedApp>
);
