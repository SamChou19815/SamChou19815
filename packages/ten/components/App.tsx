import React, { ReactElement } from 'react';

import Button from '@material-ui/core/Button';
import Head from 'next/head';

import DistributedGameCard from './DistributedGameCard';
import LocalGameCard from './LocalGameCard';
import MaterialThemedApp from './MaterialThemedApp';

type Mode = 'Local' | 'Distributed';

export default function App(): ReactElement {
  const [mode, setMode] = React.useState<Mode>('Local');

  const buttons = (
    <>
      <a className="navbar__item navbar__link" href="https://developersam.com">
        Home
      </a>
    </>
  );

  const head = (
    <Head>
      <meta charSet="utf-8" />
      <link rel="shortcut icon" href="/favicon.ico" />
      <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
      <meta name="theme-color" content="#000000" />
      <link rel="manifest" href="/manifest.json" />
      <title>TEN Game</title>
    </Head>
  );

  return (
    <MaterialThemedApp title={`TEN - ${mode}`} buttons={buttons}>
      <>
        {head}
        <div className="card">
          <div className="card__footer">
            <Button color="inherit" onClick={(): void => setMode('Local')}>
              Local
            </Button>
            <Button color="inherit" onClick={(): void => setMode('Distributed')}>
              Distributed
            </Button>
          </div>
        </div>
        {mode === 'Local' ? <LocalGameCard /> : <DistributedGameCard />}
      </>
    </MaterialThemedApp>
  );
}
