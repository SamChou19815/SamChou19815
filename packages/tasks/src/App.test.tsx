import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('App matches snapshot.', () => {
  expect(renderer.create(<App />).toJSON()).toMatchSnapshot();
});
