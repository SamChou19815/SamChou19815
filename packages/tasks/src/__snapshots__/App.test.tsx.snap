// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot. 1`] = `
<div
  className="App"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root Title MuiTypography-h6 MuiTypography-colorInherit"
      >
        Tasks
      </h6>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
            href="https://developersam.com"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Home
          </a>
        </span>
      </button>
    </div>
  </header>
  <div />
</div>
`;
